src/Experimental/traits.jl

```diff
@@ -0,0 +1,88 @@
+
+"""
+Type characterizing the mutability/immutability of a graph type.
+
+A mutable graph type `G` is expected to implement the mutability functions:
+- `add_vertex!(g::G) -> Bool`
+- `rem_vertex!(g::G, v) -> Bool`
+- `add_edge!(g::G) -> Bool`
+- `rem_edge!(g::G, i, j) -> Bool`
+- `add_vertices!(g::G, n::Integer) -> Integer` (optional)
+- `rem_vertices!(g::G, vs; keep_order::Boolean=false) -> vmap` (optional)
+"""
+abstract type GraphMutability end
```

Comment:
I think we should also call this `IsMutable`, so that we can user the constructor for this abstract type. Julia Base also does this with `Base.IteratorSize`.

Also, maybe we can have an abstract type `GraphTrait` or `AbstractGraphTrait`, then `IsMutable` and every other trait can be a subtype of this trait.

Reply:
interesting I hadn't looked at the design, I changed it in last commit